src/pages/workspaces/workspace/analysis/file-utils.test.ts

```diff
@@ -0,0 +1,113 @@
+import '@testing-library/jest-dom'
+
+import { render } from '@testing-library/react'
+import { act } from 'react-dom/test-utils'
+import { div, h } from 'react-hyperscript-helpers'
+import { AzureStorage, AzureStorageContract } from 'src/libs/ajax/AzureStorage'
+import { GoogleStorage, GoogleStorageContract } from 'src/libs/ajax/GoogleStorage'
+import { workspaceStore } from 'src/libs/state'
+import {
+  CloudProviderType,
+  cloudProviderTypes
+} from 'src/libs/workspace-utils'
+import { defaultAzureWorkspace, defaultGoogleWorkspace } from 'src/pages/workspaces/workspace/analysis/_testData/testData'
+import {
+  AbsolutePath,
+  AnalysisFile,
+  getDisplayName,
+  getExtension,
+  getFileName,
+  useAnalysisFiles
+} from 'src/pages/workspaces/workspace/analysis/file-utils'
+import { cloudProviders } from 'src/pages/workspaces/workspace/analysis/runtime-utils'
+import { getToolFromFileExtension, ToolLabel } from 'src/pages/workspaces/workspace/analysis/tool-utils'
+import { asMockedFn } from 'src/testing/test-utils'
+
+
+jest.mock('src/components/workspace-utils', () => {
+  const originalModule = jest.requireActual('src/components/workspace-utils')
+  return {
+    ...originalModule
+  }
+})
+
+jest.mock('src/libs/ajax/GoogleStorage')
+jest.mock('src/libs/ajax/AzureStorage')
+
+//TODO: test commons
+const getTestFile = (abs: AbsolutePath, cloudProvider: CloudProviderType = cloudProviderTypes[cloudProviders.gcp.label]): AnalysisFile => ({
+  name: abs,
+  ext: getExtension(abs),
+  displayName: getDisplayName(abs),
+  fileName: getFileName(abs),
+  tool: getToolFromFileExtension(getExtension(abs)) as ToolLabel,
+  lastModified: new Date().getTime(),
+  cloudProvider
+})
+
+export const HookWrapperComponent = () => {
```

Comment:
Alternatively, we have [react-hooks-testing-library](https://react-hooks-testing-library.com/) for testing hooks. See `libs/ajax/increment-response/useIncrementalResponse.test.ts` for an example.

Reply:
cool, didn't know about this lib

Reply:
That's my fault. I added it recently but didn't publicize it.

Reply:
No worries, happy there's a lib for that 😄